File: Golang/LinkedListCycleII/LinkedListCycleII_test.go
package LinkedListCycleII

import "testing"

func TestDetectCycle(t *testing.T) {
	if detectCycle(nil) == nil {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if detectCycle(&ListNode{
		Val: 0,
	}) == nil {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/LinkedListCycleIITest.py
from unittest import TestCase

from LinkedListCycleII import LinkedListCycleII, ListNode


class TestLinkedListCycleII(TestCase):
    def test_detectCycle(self):
        llc = LinkedListCycleII()

        self.assertIsNone(llc.detectCycle(None))

        self.assertIsNone(llc.detectCycle(ListNode(0)))
